lib/ShareByCircleProvider.php

```diff
@@ -155,7 +124,7 @@ public function __construct(
 		$this->userManager = $userManager;
 		$this->rootFolder = $rootFolder;
 		$this->l10n = $l10n;
-		$this->logger = $logger;
+		$this->logger = OC::$server->get(LoggerInterface::class);
```

Comment:
Any reason to not use dependency injection here?

Reply:
legacy code on core :]

Reply:
Ok, but then you don't have to pass `$logger` into `__construct()`, do you?

Reply:
i don't want to touch legacy code right now, might fix for nc26